### PR TITLE
Sample app refactor (SQLAlchemy)

### PR DESCRIPTION
**Pull request by ericharmeling**
This PR introduces the following changes:

- A new, true Hello World app that attempts a database connection and prints to the console. This app is written for SQLAlchemy, and will be the first of many.
- The existing "Hello World" apps have been renamed to more accurately reflect their function and purpose (i.e., "Simple CRUD" apps).
- A refactored Simple CRUD app this uses a `.sql` file to initialize the database, and performs inserts, selects, updates, and deletes. This app is written for SQLAlchemy, and will be the first of many.
- A refactored and renamed Sample Apps overview page, with a list of all tools, support, and available sample apps.
- Updates to the ToC, to reflect the changes.

The sample app is based on https://github.com/cockroachlabs/simple-crud-python-sqlalchemy/pull/6

### CODE REVIEW

**Comment by ericharmeling** [comment]
@vy-ton Ping on review :)

**Review by vy-ton — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10851)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling)

___
*[v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md, line 90 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV81jl-48Bj_vrGBH1:-MeV81jl-48Bj_vrGBH2:b-vg5f8t) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md#L90)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ shell
> $ git clone https://github.com/cockroachlabs/simple-crud-python-sqlalchemy/
> ```

For future proofing, I think we could call this `examples-sqlalchemy`. We can still refer to it as `Simple CRUD` in the docs
___
*[v21.1/example-apps.md, line 2 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV6oR3Be2sd-jD99My:-MeV6oR460YoVDBz3Zb3:b3qqe7e) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L2)):*
> ```Markdown
> ---
> title: Example Apps
> ```

Should we standardize on `Example Apps` or `Sample Apps`?
___
*[v21.1/example-apps.md, line 19 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV7FtB0UnHQCp3NKTG:-MeV7FtCDApl-i8kW77u:bit5k79) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L19)):*
> ```Markdown
> {{site.data.alerts.end}}
> 
> ## Python
> ```

Would be nice to have a consistent order of languages with the sidebar.
___
*[v21.1/example-apps.md, line 72 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV6alj8m0xhXLiwBLD:-MeV6alj8m0xhXLiwBLE:b2z4s8p) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L72)):*
> ```Markdown
> | [libpqxx](https://github.com/jtv/libpqxx)     | Beta           | [Simple CRUD](build-a-c++-app-with-cockroachdb.html)
> 
> ## PHP
> ```

Since we moved PHP and Clojure to community tooling, I would be in favor of removing them here
___
*[v21.1/hello-world-python-sqlalchemy.md, line 36 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV7kLH4m-X7aPgsfx6:-MeV7kLH4m-X7aPgsfx7:b-g0fric) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/hello-world-python-sqlalchemy.md#L36)):*
> ```Markdown
> {% include copy-clipboard.html %}
> ~~~ shell
> $ git clone https://github.com/cockroachlabs/hello-world-python-sqlalchemy/
> ```

`hello-world-sqlalchemy`

**Review by ericharmeling — COMMENTED** [review]
TFTR @vy-ton !

I think I've gotten to all of your comments. I replied to a couple, with additional thoughts/questions. PTAL

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10851#-:-MezEJc19iqwUO9ht_r0:b-u0g5i4)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans and @vy-ton)

___
*[v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md, line 90 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV81jl-48Bj_vrGBH1:-MezIs8F7n-2sY6FsIN3:b-zdwg4g) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md#L90)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

For future proofing, I think we could call this `examples-sqlalchemy`. We can still refer to it as `Simple CRUD` in the docs
</blockquote></details>

So there are two things we are trying to do at the same time with this repo:

1. Provide a cloneable, forkable repo for users who want to run and/or build on a simple, but functional, example app. This is what we currently use our example app repos for, and is the original purpose of the repos.

2. Use a GitHub repo to store functional snippets of code that we want to include in our docs. We currently don't really do this, but it is in the plan for the future (as we have discussed offline).

Because of the **1**, I'm hesitant to name the repo after the code snippets that we might pull from it. The code runs as a single example app (and likely always will), we currently document the app as a single example app, and users cloning/forking the repo will likely be running the entire app.

Because of **2** (and to your point re: future proofing) - as we expand the app (possibly to include more than just "Simple CRUD" functionality) for the sake of having more examples/snippets that we can pull from this repo in "recipe docs", the code could outgrow the current name and purpose.

So ... it might make most sense to name the repo `example-app-python-sqlalchemy`. (As I note below, and as I've mentioned in this PR's corresponding Google Doc, the naming convention remains `app-language-driver`. ) 

It is going to be hard to do both **1** and **2** at the same time with a single repo, but IMO this is something we will have to navigate in the next iteration of our example app refactoring.

Thoughts?
___
*[v21.1/example-apps.md, line 2 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV6oR3Be2sd-jD99My:-MezEY4502EpNjvw1SbH:bt9gg2l) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L2)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

Should we standardize on `Example Apps` or `Sample Apps`?
</blockquote></details>

I have updated the docs to be consistent with "Example Apps".
___
*[v21.1/example-apps.md, line 19 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV7FtB0UnHQCp3NKTG:-MezIxSM4P-cH0nEfOCU:b-ast2ue) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L19)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

Would be nice to have a consistent order of languages with the sidebar.
</blockquote></details>

I agree. I've updated this page and the "Example Apps" ToC to be consistent.

IMO, this is a bigger issue. We list out our tools on other pages as well, with orderings inconsistent with the ToC. 

FWIW, I am of the opinion that we should make flat language/tooling listings/ToCs alphabetical. Developers look for their language/tool where they expect to see it (i.e., alphabetically/categorically), not according to our support levels, which they may or may not be aware of.

@ianjevans To consider as you work on the nav restructure.
___
*[v21.1/example-apps.md, line 72 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV6alj8m0xhXLiwBLD:-MezEjf-AFYm5PMxxOSw:bwwzj94) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L72)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

Since we moved PHP and Clojure to community tooling, I would be in favor of removing them here
</blockquote></details>

I've updated the "Support Level" of these tools to be "Community" (and not "Beta"). However, I do think we should include all example apps that are listed in the ToC here, even if they are based on community-supported tools. It would be pretty confusing to see languages in the ToC under "Example Apps", and not see those apps on this page. 

I did update the note at the top to include a couple sentences about community support. Also, the Rust example app (also in the ToC) was missing - I added it in here too, with the "Community" support-level. 

... One alternative would be to remove the apps for community-supported tools from our docs altogether. I am not necessarily opposed to this, but it might be a little unfair to our PHP/Rust users.
___
*[v21.1/hello-world-python-sqlalchemy.md, line 36 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV7kLH4m-X7aPgsfx6:-MezEoVvAK6B9nx8SRnf:b-aob303) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/hello-world-python-sqlalchemy.md#L36)):*
<details><summary><i>Previously, vy-ton (Vy Ton) wrote…</i></summary><blockquote>

`hello-world-sqlalchemy`
</blockquote></details>

The actual repo name is [`hello-world-python-sqlalchemy`](https://github.com/cockroachlabs/hello-world-python-sqlalchemy). This name is the same as before, and it is consistent with the format of the other hello world repo names (e.g., [`hello-world-ruby-activerecord`](https://github.com/cockroachlabs/hello-world-ruby-activerecord)). I've commented on the Google Doc on this topic as well.

**Review by ericharmeling — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10851)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans and @vy-ton)

___
*[v21.1/example-apps.md, line 72 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV6alj8m0xhXLiwBLD:-MezoTgd-6PxAllDEohS:b-8923cj) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L72)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

I've updated the "Support Level" of these tools to be "Community" (and not "Beta"). However, I do think we should include all example apps that are listed in the ToC here, even if they are based on community-supported tools. It would be pretty confusing to see languages in the ToC under "Example Apps", and not see those apps on this page. 

I did update the note at the top to include a couple sentences about community support. Also, the Rust example app (also in the ToC) was missing - I added it in here too, with the "Community" support-level. 

... One alternative would be to remove the apps for community-supported tools from our docs altogether. I am not necessarily opposed to this, but it might be a little unfair to our PHP/Rust users.
</blockquote></details>

Also, FWIW, the tentative plan is to rewrite the community-supported tool example apps to just be Hello World apps.

**Review by vy-ton — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10851)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @ianjevans, and @vy-ton)

___
*[v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md, line 90 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV81jl-48Bj_vrGBH1:-Mezt8uP8g_NYU__rI-k:b-m8gfra) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md#L90)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

So there are two things we are trying to do at the same time with this repo:

1. Provide a cloneable, forkable repo for users who want to run and/or build on a simple, but functional, example app. This is what we currently use our example app repos for, and is the original purpose of the repos.

2. Use a GitHub repo to store functional snippets of code that we want to include in our docs. We currently don't really do this, but it is in the plan for the future (as we have discussed offline).

Because of the **1**, I'm hesitant to name the repo after the code snippets that we might pull from it. The code runs as a single example app (and likely always will), we currently document the app as a single example app, and users cloning/forking the repo will likely be running the entire app.

Because of **2** (and to your point re: future proofing) - as we expand the app (possibly to include more than just "Simple CRUD" functionality) for the sake of having more examples/snippets that we can pull from this repo in "recipe docs", the code could outgrow the current name and purpose.

So ... it might make most sense to name the repo `example-app-python-sqlalchemy`. (As I note below, and as I've mentioned in this PR's corresponding Google Doc, the naming convention remains `app-language-driver`. ) 

It is going to be hard to do both **1** and **2** at the same time with a single repo, but IMO this is something we will have to navigate in the next iteration of our example app refactoring.

Thoughts?
</blockquote></details>

Naming suggestion SGTM.

Hopefully, we can achieve (2) as methods within the functional, example app. For recipes, we still want users to be able to run/build the recipe.
___
*[v21.1/example-apps.md, line 19 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV7FtB0UnHQCp3NKTG:-Meztk1V6zca7eVg4mMY:b-42j2j4) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L19)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

I agree. I've updated this page and the "Example Apps" ToC to be consistent.

IMO, this is a bigger issue. We list out our tools on other pages as well, with orderings inconsistent with the ToC. 

FWIW, I am of the opinion that we should make flat language/tooling listings/ToCs alphabetical. Developers look for their language/tool where they expect to see it (i.e., alphabetically/categorically), not according to our support levels, which they may or may not be aware of.

@ianjevans To consider as you work on the nav restructure.
</blockquote></details>

+1 for alphabetical ordering
___
*[v21.1/example-apps.md, line 72 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV6alj8m0xhXLiwBLD:-MeztuV12ZG0soMljsI8:bs7xwkh) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L72)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

Also, FWIW, the tentative plan is to rewrite the community-supported tool example apps to just be Hello World apps.
</blockquote></details>

I'm good with including them, we should just prioritize their refactoring lower than other languages/tools. Going forward, we should have a `Hello world` app for any tool we maintain support for. For additional apps, we will likely need to be more protective of our time/resources and prioritize only some

**Review by ericharmeling — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10851)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans and @vy-ton)

___
*[v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md, line 90 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV81jl-48Bj_vrGBH1:-Mf-AFQd-UlgqirVlwbp:bh1xqw9) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md#L90)):*
>Naming suggestion SGTM.

Great. Updated this docs PR. Will update the live repo name after this gets writer approval for a merge.

> Hopefully, we can achieve (2) as methods within the functional, example app. For recipes, we still want users to be able to run/build the recipe.

+1
___
*[v21.1/example-apps.md, line 72 at r1](https://reviewable.io/reviews/cockroachdb/docs/10851#-MeV6alj8m0xhXLiwBLD:-Mf-9hn-9QApCKqvZQmq:b-clakoz) ([raw file](https://github.com/cockroachdb/docs/blob/6b2da3583803f67cba1fcd6b217d7921fd08e732/v21.1/example-apps.md#L72)):*
> we should just prioritize their refactoring lower than other languages/tools

Agreed!

File: v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md [hunk 1]
```diff
@@ -43,133 +35,175 @@ For other ways to install SQLAlchemy, see the [official documentation](http://do
...
+</section>
+
+## Step 3. Get the code
+
+Clone the code's GitHub repo:
 
 {% include copy-clipboard.html %}
 ~~~ shell
-$ git clone https://github.com/cockroachlabs/simple-crud-python-sqlalchemy/
+$ git clone https://github.com/cockroachlabs/example-app-python-sqlalchemy/
```
Comment (ianjevans) [inline v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md:90]: Same as above.

File: v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md [hunk 2]
```diff
@@ -43,133 +35,175 @@ For other ways to install SQLAlchemy, see the [official documentation](http://do
...
+├── dbinit.sql
+├── main.py
+└── models.py
+~~~
+
+The `dbinit.sql` file initializes the database schema that the application uses:
+
+{% include_cached copy-clipboard.html %}
+~~~ python
+{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-python-sqlalchemy/master/dbinit.sql %}
```
Comment (ianjevans) [inline v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md:106]: Same as above.

File: v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md [hunk 3]
```diff
@@ -43,133 +35,175 @@ For other ways to install SQLAlchemy, see the [official documentation](http://do
...
+~~~
+
+The `models.py` uses SQLAlchemy to map the `Accounts` table to a Python object:
+
+{% include_cached copy-clipboard.html %}
 ~~~ python
-{% include {{page.version.version}}/app/python/sqlalchemy/example.py %}
+{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-python-sqlalchemy/master/models.py %}
```
Comment (ianjevans) [inline v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md:113]: Same as above.

File: v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md [hunk 4]
```diff
@@ -43,133 +35,175 @@ For other ways to install SQLAlchemy, see the [official documentation](http://do
...
 ~~~
 
-### Update the connection parameters
+The `main.py` uses SQLAlchemy to map Python methods to SQL operations:
 
-In the `create_engine()` function, update the connection string as follows:
+{% include_cached copy-clipboard.html %}
+~~~ python
+{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-python-sqlalchemy/master/main.py %}
```
Comment (ianjevans) [inline v21.1/build-a-python-app-with-cockroachdb-sqlalchemy.md:120]: Same as above.

**Review by ianjevans — APPROVED** [review]
I don't know if the intent is to rename `simple-crud-python-sqlalchemy`, but I flagged all the places where the repo is `example-crud-python-sqlalchemy` as that repo doesn't exist.

**Review by ericharmeling — COMMENTED** [review]
> I don't know if the intent is to rename simple-crud-python-sqlalchemy

That's exactly right. 

See my earlier comment (in my review thread with Vy):

> Great. Updated this docs PR. Will update the live repo name after this gets writer approval for a merge.

After this is approved, I will rename the repo, and then merge these changes.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10851#-:-Mf91p1w2VHFwCWkyrJG:b33d29k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @ianjevans, and @vy-ton)

**Comment by ericharmeling** [comment]
@ianjevans TFTR!

Just renamed the repo: https://github.com/cockroachlabs/example-app-python-sqlalchemy

Going to merge.

**Pull request merged**